imap/monitor.go

```diff
@@ -192,3 +200,45 @@ func checkForNewEmails(im models.IMAP) {
 		log.Debug("No new emails for ", im.Username)
 	}
 }
+
+// returns a slice of gophish rid paramters found in the email HTML, Text, and attachments
+func checkRIDs(em *email.Email) (map[string]int, error) {
+
+	rids := make(map[string]int)
+
+	// Check Text and HTML
+	emailContent := string(em.Text) + string(em.HTML)
+	for _, r := range goPhishRegex.FindAllStringSubmatch(emailContent, -1) {
+		newrid := r[len(r)-1]
+		if _, ok := rids[newrid]; ok {
```

Comment:
Small nit: we can make this a bit cleaner. Something like:

```go
rid := r[len(r)-1]
if _, ok := rids[newrid]; !ok {
    rids[newrid] = 0
}
rids[newrid]++
```

Reply:
Ah, much neater pattern, thanks!